modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -375,34 +375,42 @@ def to_numpy(self, **kwargs):
     # such that columns/rows that don't have an index on the other DataFrame
     # result in NaN values.
 
-    add = Binary.register(pandas.DataFrame.add)
-    combine = Binary.register(pandas.DataFrame.combine)
-    combine_first = Binary.register(pandas.DataFrame.combine_first)
-    eq = Binary.register(pandas.DataFrame.eq)
-    floordiv = Binary.register(pandas.DataFrame.floordiv)
-    ge = Binary.register(pandas.DataFrame.ge)
-    gt = Binary.register(pandas.DataFrame.gt)
-    le = Binary.register(pandas.DataFrame.le)
-    lt = Binary.register(pandas.DataFrame.lt)
-    mod = Binary.register(pandas.DataFrame.mod)
-    mul = Binary.register(pandas.DataFrame.mul)
-    rmul = Binary.register(pandas.DataFrame.rmul)
-    ne = Binary.register(pandas.DataFrame.ne)
-    pow = Binary.register(pandas.DataFrame.pow)
-    radd = Binary.register(pandas.DataFrame.radd)
-    rfloordiv = Binary.register(pandas.DataFrame.rfloordiv)
-    rmod = Binary.register(pandas.DataFrame.rmod)
-    rpow = Binary.register(pandas.DataFrame.rpow)
-    rsub = Binary.register(pandas.DataFrame.rsub)
-    rtruediv = Binary.register(pandas.DataFrame.rtruediv)
-    sub = Binary.register(pandas.DataFrame.sub)
-    truediv = Binary.register(pandas.DataFrame.truediv)
-    __and__ = Binary.register(pandas.DataFrame.__and__)
-    __or__ = Binary.register(pandas.DataFrame.__or__)
-    __rand__ = Binary.register(pandas.DataFrame.__rand__)
-    __ror__ = Binary.register(pandas.DataFrame.__ror__)
-    __rxor__ = Binary.register(pandas.DataFrame.__rxor__)
-    __xor__ = Binary.register(pandas.DataFrame.__xor__)
+    add = Binary.register(pandas.DataFrame.add, how_compute_dtypes="common_cast")
+    combine = Binary.register(
+        pandas.DataFrame.combine, how_compute_dtypes="common_cast"
+    )
+    combine_first = Binary.register(
+        pandas.DataFrame.combine_first, how_compute_dtypes="bool"
+    )
+    eq = Binary.register(pandas.DataFrame.eq, how_compute_dtypes="bool")
+    floordiv = Binary.register(
+        pandas.DataFrame.floordiv, how_compute_dtypes="common_cast"
+    )
+    ge = Binary.register(pandas.DataFrame.ge, how_compute_dtypes="bool")
```

Comment:
Please check the API layer methods where we pass dtypes in to the qc and remove those arguments. We should consolidate dtypes inferreng logic in one place. As an example, https://github.com/modin-project/modin/blob/843b0cf02df9ff76f32cc75749203762633ffb27/modin/pandas/base.py#L1409.

Reply:
Done.